cmd/bosun/sched/check.go

```diff
@@ -244,10 +245,34 @@ func (s *Schedule) runHistory(r *RunHistory, ak models.AlertKey, event *models.E
 		if a.Log || silencedOrIgnored(a, event, si) {
 			//a log or silenced/ignored alert will not need to be saved
 		} else {
-			incident.Id, err = s.DataAccess.State().UpdateIncidentState(incident)
+			daState := s.DataAccess.State()
+			incident.Id, err = daState.UpdateIncidentState(incident)
 			if err != nil {
 				return
 			}
+			previousIds := []int64{}
+			previousIds, err = daState.GetAllIncidentIdsByAlertKey(ak)
+			if err != nil {
+				return
+			}
+			for _, id := range previousIds {
+				if incident.Id > id {
+					incident.PreviousIds = append(incident.PreviousIds, id)
```

Comment:
Wait, we're storing **all** previous alert keys on **every** alert? Gross. Why?

Reply:
@captncraig To make it easier to either:
 1. Display more than just the previous incident in a view
 2. Make it easier to iterate over all previous incidents in templates without having to follow a linked list.

Reply:
Also to clarify not the previous alert keys, just an array of incident id numbers (int64)

Reply:
Another Also :P alert keys are the alert name plus the tagset (not all alerts under the name, unless the tagset is `{}` or of len 1. So the list generally should not be crazy long.

Reply:
I guess I'm just uncomfortable anytime I see potentially large, and also redundant, and also repetitive collections or lists. 

I guess though I'd rather a template could show them all without having to recursively traverse the chain, which was gonna be the alternative I suggested. So ok.